.circleci/config.yml

```diff
@@ -144,8 +144,11 @@ jobs:
         type: "string"
 
     docker:
-      # Run in a highly Nix-capable environment.
-      - image: "nixos/nix:latest"
+      # Run in a highly Nix-capable environment.  We used to use `latest` but
+      # one day someone pushed a bad revision to it and our CI broke.  So now
+      # we just pin some recent version.  Who would have thought a floating
+      # dependency would cause build instability?
+      - image: "nixos/nix:2.3"
```

Comment:
This doesn't seem like a particularly recent version. Looking at https://hub.docker.com/r/nixos/nix/tags suggests 2.3.12 or 2.3.16 as recent version. It looks like the new images are built with nix, but the older ones are based on https://github.com/NixOS/docker which appears to be apline with nix installed. I think both the versions I mention are that old style.